File: go/2-variables.go
package main
import "fmt"
// comment

/* 
 * long comment
 * on multiple lines
 */

func main() {
    var str string ="hello"
    var number int =123

    var char uint8 ='a'
    var str1="world"
    const constant int =10
    a:=11.10

    fmt.Println(number)
    fmt.Println(str1)
    fmt.Println(constant)
    fmt.Println(a)
    fmt.Printf("types: number=%T char=%T str=%T a=%T\n", number, char, str, a);
    fmt.Printf("number=%d char=%c char=%d str=%s a=%f\n", number, char, char, str, a);

    var pointer *int = &number
    var truth = true
    fmt.Printf("truth type is %T\n", truth)
	
		const (
		january = iota
		february
		may=5
		june =iota
	)
	
	fmt.Printf("string=%s number=%d char=%c pointer=%x value at pointer=%d\n", str, number, char, pointer, *pointer);
	*pointer = 456
	fmt.Printf("string=%s number=%d char=%c pointer=%p value at pointer=%d\n", str, number, char, pointer, *pointer);
	
	s:=january
	f:=february
	m:= may
	j:= june
	
	fmt.Printf("months:  january=%d february=%d may=%d june=%d\n",s, f, m, j)
}

File: perl/2-variables.pl
#!/bin/perl
# comment

=begin
 * long comment
 * on multiple lines
=cut

    $string ="hello \$ \" ";
    $string2="world";

    print $string . $string2;
    print "\n";
    print "$string" . "$string2 \n";

    $str1 = <<"EOF";
long text
second line
EOF

$number=123;
$char='a';
@array=(4, 5, 6);
%hash=('key', 10, 'key1', 'value');
$foo=123;
@foo=("ala", "ma", "kota");
%foo=('key', 'value');

print "number=$number char=$char string=$string str1=$str1 array[0]=$array[0] hash{'key'}=$hash{'key'}\n";
print "\$foo=$foo \@foo=@foo \$foo[0]=$foo[0] \%foo{'key'}=$foo{'key'}\n";
$hex=0xff;
printf "hex=$hex";

$smile  = v9786;
$foo    = v102.111.111;
$martin = v77.97.114.116.105.110; 
$char_a = v97;
$char_a_hex = "\x{61}";
$chars_abc = v97.98.99;

print "smile = $smile\n";
print "foo = $foo\n";
print "martin = $martin\n";
print "char_a = $char_a\n";
print "char_a_hex = $char_a_hex\n";
print "chars_abc = $chars_abc\n";

print "File name ". __FILE__ . "\n";
print "Line Number " . __LINE__ ."\n";
print "Package " . __PACKAGE__ ."\n";

$ref=\$number;
print "number=$number ref=$ref value=$$ref\n";


